core/src/main/java/org/apache/struts2/interceptor/csp/CspInterceptor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.interceptor.csp;
+
+import com.opensymphony.xwork2.ActionInvocation;
+import com.opensymphony.xwork2.interceptor.AbstractInterceptor;
+import com.opensymphony.xwork2.interceptor.PreResultListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Interceptor that implements Content Security Policy on incoming requests used to protect against
+ * common XSS and data injection attacks. Uses {@link CspSettings} to filter the requests allowed to be processed.
+ *
+ * @see <a href="https://csp.withgoogle.com/docs/index.html">https://csp.withgoogle.com/docs/index.html/</a>
+ **/
+public class CspInterceptor extends AbstractInterceptor implements PreResultListener {
+
+    private boolean enforcingMode = false;
+    private CspSettings settings = new DefaultCspSettings();
+
+    public void setReportUri(String reportUri) {
+        settings.setReportUri(reportUri);
+    }
+
+    public void setEnforcingMode(String value){
+        this.enforcingMode = Boolean.parseBoolean(value);
+        this.settings.setEnforcingMode(this.enforcingMode);
```

Comment:
remove the `this` keyword from all these variables
```suggestion
        enforcingMode = Boolean.parseBoolean(value);
        settings.setEnforcingMode(enforcingMode);
```